code/script.js

```diff
@@ -269,57 +297,117 @@ const filterCharacters = (keep) => {
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
+// store the interaction from the player in a variable.
   // If the player wants to guess, invoke the checkMyGuess function.
+const guess = (personToConfirm) => {
+  let guessed = confirm (`hmmmm are you sure?, really really sure that ${personToConfirm} is your guess?`)
+  if(guessed){
+    let personToCheck = personToConfirm; //variable that saves the character the player has shosen
+checkMyGuess(personToCheck)
+
+  }
+  //if no
+  else {
+    alert (' when in doubt... Lets continue instead')
+  }
+ 
 }
 
 // If you confirm, this function is invoked
+
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name){
+    
+    winOrLoseText.innerHTML = `${secret.name} Que vivaaaaaa, arribaaaa hurraaaay its totaly correct!
+  }<br>Well done! <br><img class="secret-revealed-img" src="${secret.img}"/>`;
+
+  // Show the win or lose section
+  winOrLose.style.display = 'flex'
+  //  Hide the game board
+  board.style.display = 'none'
+ 
+}
+else {
+  winOrLoseText.innerHTML = `Ay nooooo, Im sorry better luck next time 😣. 
+  <br> Right answer was ${secret.name}.<br>
+  
+  `;
+  // Show the win or lose section
+  winOrLose.style.display = 'flex'
+  //  Hide the game board
+  board.style.display = 'none';
+  
+}
 }
 
 // Invokes the start function when website is loaded
 start()
 
-// All the event listeners
-restartButton.addEventListener('click', start)
+//This makes the site reload when player clicks on "try again" 
+const refreshPage = () => {
+  //Hide the winOrLose and show generate board... 
+  winOrLose.style.display = 'none';
+  //  Hide the game board
+  board.style.display = 'flex';
+  start();
+}
+const playAgain = ()=>{
+  start 
+
+}
+
+
+restartButton.addEventListener('click', start);
+
+questions.addEventListener('change', selectQuestion); //this calls the selectQuestion function as soon as the player changes option in dropdown.
+findOutButton.addEventListener('click', () => {
+  checkQuestion()
+});//Is there another way to do this??
+
+playAgainButton.addEventListener ('click', playAgain) //seems like this wont work for me!
+
```

Comment:
Not working. Try this:

playAgain.addEventListener("click", (event) => {
  start();
 /*  pause();
  reset(); */
  winOrLose.style.display = "none";
  window.location.reload();
});